tests/test_runtime.py

```diff
@@ -120,6 +120,21 @@ def test_double_stop(self):
         self.runtime.wait_completed(sync=True)
         assert self.runtime.is_stopped
 
+    @pytest.mark.asyncio
+    async def test_runtime_cancelled(self):
+        """Test runtime cancelled and stopped properly."""
+        self.runtime.set_loop(asyncio.get_event_loop())
+        task = asyncio.ensure_future(self.runtime.run_runtime())
+        await wait_for_condition_async(lambda: self.runtime.is_running, timeout=5)
+        assert self.runtime.agent_loop.is_running
+        task.cancel()
+        await asyncio.sleep(0.1)
+
+        with pytest.raises(asyncio.CancelledError):
+            await task
+
+        assert not self.runtime.agent_loop.is_running
```

Comment:
just curious since you said we should remove all sleep. How would you remove it here? I'm thinking we could combine `wait_for_condition` pattern with a signal handler. Is there a better or easier way?

Reply:
@Karrenbelt
my bad. it ws experimental thing during unreachable code  investigation. forgot to remove

Reply:
done